lib/services/singleton_manager.ts

```diff
@@ -186,73 +187,68 @@ export class SNSingletonManager extends PureService {
       return a.created_at < b.created_at ? -1 : 1;
     });
     const deleteItems = arrayByRemovingFromIndex(earliestFirst, 0);
-    await this.itemManager!.setItemsToBeDeleted(Uuids(deleteItems));
+    await this.itemManager.setItemsToBeDeleted(Uuids(deleteItems));
   }
 
-  public async findOrCreateSingleton(
+  public async findOrCreateSingleton<T extends SNItem = SNItem>(
     predicate: SNPredicate,
     createContentType: ContentType,
     createContent: PayloadContent
-  ) {
-    // eslint-disable-next-line no-async-promise-executor
-    return new Promise(async (resolve) => {
-      const matchingItems = this.validItemsMatchingPredicate(predicate);
-      if (matchingItems.length > 0) {
-        return resolve(matchingItems[0]);
-      }
-      /** Item not found, safe to create after full sync has completed */
```

Comment:
This comment didn't make it in the new code, I think we want to retain it. This logic can be intricate so we want to make sure we have as much context as possible.